pkg/controller/core/core.go

```diff
@@ -83,6 +83,7 @@ func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache
 	).SetupWithManager(mgr, cfg); err != nil {
 		return "Workload", err
 	}
+	qManager.AddTopologyUpdateWatcher(cqRec)
```

Comment:
Can we add topology watcher in the resonciler so that we can avoid to take the qManager lock?

Reply:
We don't need to the qManager lock, because the watchers are only added during startup.

I tried initially in reconciler, but this is problematic, because it would mean registering TAS RF reconciler in the CQ core reconciler, but this breaks the dependency - because currently TAS controllers are in a dedicated package. TAS reconcilers depend on core reconcilers, but not the other way round. So, placing the watchers in qManager breaks the dependency issue. I think we could use this pattern for other watchers. I think it works conceptually.

Reply:
Uhm, I see. In that case, I think that we should restructure the package for the TAS so that we can distinguish the responsibility between each component.

But based on the release schedule, I'm ok with this implementation for now.